scripts/portfolio.js

```diff
@@ -0,0 +1,50 @@
+(function(module) {
+
+  function Project (opts) {
+    this.author = opts.author;
+    this.title = opts.title;
+    this.category = opts.category;
+    this.authorUrl = opts.authorUrl;
+    this.publishedOn = opts.publishedOn;
+    this.body = opts.body;
+  }
+
+  Project.all = [];
+
+  Project.prototype.toHtml = function() {
+    var source = $('#project-template').html();
+    var template = Handlebars.compile(source);
+
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+
+    return template(this);
+  };
+
+  Project.loadAll = function(myData) {
+    myData.sort(function(a,b) {
+      return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+    });
+
+    Project.all = myData.map(function(ele) {
+      return new Project(ele);
+    });
+  };
+
+  Project.grabAll = function(callback) {
+    if(localStorage.myData) {
+      Project.loadAll(JSON.parse(localStorage.myData));
+      callback();
+    } else {
+      $.getJSON('/data/portfolioArticles.json', function(myData) {
+        Project.loadAll(myData);
+
+        var toCache = JSON.stringify(myData);
+        localStorage.myData = toCache;
+      });
+      callback();
```

Comment:
Haven't pushed my updated files yet but they will include the change we discussed here. I believe that this also fixed my category filter as well, which for some reason wasn't pulling data properly.